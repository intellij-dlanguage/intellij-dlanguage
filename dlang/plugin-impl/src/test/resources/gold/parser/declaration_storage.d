struct S
{
    int x;
    int method() const
    {
	//x++;    // Error: this method is const and cannot modify this.x
        return x; // OK: we can still read this.x
    }
}