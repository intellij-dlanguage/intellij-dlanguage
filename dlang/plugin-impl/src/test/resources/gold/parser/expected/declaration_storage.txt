D Language File
  DLanguageDeclarationImpl(DECLARATION)
    STRUCT_DECLARATION
      PsiElement(DlangTokenType.struct)('struct')
      IDENTIFIER
        PsiElement(DlangTokenType.ID)('S')
      DLanguageStructBodyImpl(STRUCT_BODY)
        PsiElement(DlangTokenType.{)('{')
        DLanguageDeclarationImpl(DECLARATION)
          DLanguageVariableDeclarationImpl(VARIABLE_DECLARATION)
            DLanguageTypeImpl(TYPE)
              DLanguageType_2Impl(TYPE_2)
                DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                  PsiElement(DlangTokenType.int)('int')
            DECLARATOR
              IDENTIFIER
                PsiElement(DlangTokenType.ID)('x')
            PsiElement(DlangTokenType.;)(';')
        DLanguageDeclarationImpl(DECLARATION)
          FUNCTION_DECLARATION
            DLanguageTypeImpl(TYPE)
              DLanguageType_2Impl(TYPE_2)
                DLanguageBuiltinTypeImpl(BUILTIN_TYPE)
                  PsiElement(DlangTokenType.int)('int')
            IDENTIFIER
              PsiElement(DlangTokenType.ID)('method')
            DLanguageParametersImpl(PARAMETERS)
              PsiElement(DlangTokenType.()('(')
              PsiElement(DlangTokenType.))(')')
            DLanguageMemberFunctionAttributeImpl(MEMBER_FUNCTION_ATTRIBUTE)
              PsiElement(DlangTokenType.const)('const')
            DLanguageFunctionBodyImpl(FUNCTION_BODY)
              DLanguageSpecifiedFunctionBodyImpl(SPECIFIED_FUNCTION_BODY)
                DLanguageBlockStatementImpl(BLOCK_STATEMENT)
                  PsiElement(DlangTokenType.{)('{')
                  PsiComment(DlangTokenType.LINE_COMMENT)('//x++;    // Error: this method is const and cannot modify this.x')
                  DLanguageDeclarationsAndStatementsImpl(DECLARATIONS_AND_STATEMENTS)
                    DLanguageDeclarationOrStatementImpl(DECLARATION_OR_STATEMENT)
                      DLanguageStatementImpl(STATEMENT)
                        DLanguageStatementNoCaseNoDefaultImpl(STATEMENT_NO_CASE_NO_DEFAULT)
                          DLanguageReturnStatementImpl(RETURN_STATEMENT)
                            PsiElement(DlangTokenType.return)('return')
                            DLanguageExpressionImpl(EXPRESSION)
                              DLanguageAssignExpressionImpl(ASSIGN_EXPRESSION)
                                DLanguageCmpExpressionImpl(CMP_EXPRESSION)
                                  DLanguageUnaryExpressionImpl(UNARY_EXPRESSION)
                                    DLanguagePrimaryExpressionImpl(PRIMARY_EXPRESSION)
                                      DLanguageIdentifierOrTemplateInstanceImpl(IDENTIFIER_OR_TEMPLATE_INSTANCE)
                                        IDENTIFIER
                                          PsiElement(DlangTokenType.ID)('x')
                            PsiElement(DlangTokenType.;)(';')
                  PsiComment(DlangTokenType.LINE_COMMENT)('// OK: we can still read this.x')
                  PsiElement(DlangTokenType.})('}')
        PsiElement(DlangTokenType.})('}')